pkg/controller/servicemesh/controlplane/controller.go

```diff
@@ -84,28 +83,13 @@ func add(mgr manager.Manager, r *ControlPlaneReconciler) error {
 	}
 
 	// watch created resources for use in synchronizing ready status
-	if err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}},
-		&handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &v2.ServiceMeshControlPlane{},
-		},
-		ownedResourcePredicates); err != nil {
+	if err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, enqueueRequestForSMCP, ownedResourcePredicates); err != nil {
 		return err
 	}
-	if err = c.Watch(&source.Kind{Type: &appsv1.StatefulSet{}},
-		&handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &v2.ServiceMeshControlPlane{},
-		},
-		ownedResourcePredicates); err != nil {
+	if err = c.Watch(&source.Kind{Type: &appsv1.StatefulSet{}}, enqueueRequestForSMCP, ownedResourcePredicates); err != nil {
 		return err
 	}
-	if err = c.Watch(&source.Kind{Type: &appsv1.DaemonSet{}},
-		&handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &v2.ServiceMeshControlPlane{},
-		},
-		ownedResourcePredicates); err != nil {
+	if err = c.Watch(&source.Kind{Type: &appsv1.DaemonSet{}}, enqueueRequestForSMCP, ownedResourcePredicates); err != nil {
```

Comment:
is this going to do anything now? CNI daemonset isn't tied to any SMCP

Reply:
tbh I'm not sure this did ever work...?

Reply:
I think CNI status is calculated elsewhere, as it's not part of the actual control plane installation.  This code would only check daemon sets that were installed through the charts.  As there are none, It can probably be safely removed at some point in the future.  For now, it's just an extra watch.